routes/drones.js

```diff
@@ -1,36 +1,105 @@
 const express = require('express');
 const router = express.Router();
 
-// require the Drone model here
+// require the Drone model 
+
+const Drone = require("../models/Drone.model");
 
 router.get('/drones', (req, res, next) => {
   // Iteration #2: List the drones
-  // ... your code here
+  Drone.find()
+    .then( (dronesFromDB) => {
+      const data = {
+        dronesArr: dronesFromDB
+      };
+      res.render("drones/list", data)
+    })
+    .catch( (error) => {
+      console.log("Error getting data from DB", error);
+      next(error);
+    })
 });
 
 router.get('/drones/create', (req, res, next) => {
   // Iteration #3: Add a new drone
-  // ... your code here
+  router.get("/drones/create", (req, res) => {
+    res.render("drones/drone-create");
+  })
 });
 
 router.post('/drones/create', (req, res, next) => {
   // Iteration #3: Add a new drone
-  // ... your code here
+
+  const drones = {
+    name: req.body.name,
+    propellers: req.body.propellers,
+    maxSpeed: req.body.MaxSpeed,
+  };
+
+  Drone.create(drones)
+    .then( () => {
+      res.redirect("/drones");
+    })
+    .catch( (error) => {
+      console.log("Error creating drone in the DB", error);
+      next(error);
+    })
 });
 
 router.get('/drones/:id/edit', (req, res, next) => {
   // Iteration #4: Update the drone
-  // ... your code here
+  const {droneId} = req.params;
```

Comment:
```
router.get('/drones/:id/edit', (req, res, next) => {
  const {droneId} = req.params;
```

` const {droneId} = req.params;` here you should have used `const { id } = req.params;` because you're using the the params as `/:id` and not `/:droneId`

This is the same in other places on the code